test/CMakeLists.txt

```diff
@@ -20,9 +20,19 @@ enable_testing()
 
 if(MSVC)
   # VS needs -C config to run correct
-  add_custom_target(tests "${CMAKE_CTEST_COMMAND}" -C $<$<CONFIG:Debug>:Debug>$<$<CONFIG:Release>:Release> -E gpu_octree -V -T Test VERBATIM)
+  if (PCL_DISABLE_GPU_TESTS)
+    # tests from gpu module excluded since CI does not have access to GPU hardware
+    add_custom_target(tests "${CMAKE_CTEST_COMMAND}" -C $<$<CONFIG:Debug>:Debug>$<$<CONFIG:Release>:Release> -E gpu -V -T Test VERBATIM)
+  else()
+    add_custom_target(tests "${CMAKE_CTEST_COMMAND}" -C $<$<CONFIG:Debug>:Debug>$<$<CONFIG:Release>:Release> -V -T Test VERBATIM)
+  endif()
 else()
-  add_custom_target(tests "${CMAKE_CTEST_COMMAND}" -E gpu_octree -V -T Test VERBATIM)
+  if (PCL_DISABLE_GPU_TESTS)
+    # tests from gpu module excluded since CI does not have access to GPU hardware
+    add_custom_target(tests "${CMAKE_CTEST_COMMAND}" -E gpu -V -T Test VERBATIM)
+  else()
+    add_custom_target(tests "${CMAKE_CTEST_COMMAND}" -V -T Test VERBATIM)
+  endif()
```

Comment:
You should have a quick check at generator expressions to avoid all these branches
https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#conditional-expressions

```cmake
$<PCL_DISABLE_GPU_TESTS:-E gpu_octree>
```